src/DataBase/UserDatabase.cs

```diff
@@ -43,36 +51,60 @@ public static int Authenticate(string email, string pass)
             return user.Id;
         }
 
-        public static bool ChangePassword(string email, string pass)
+        public async Task<bool> ChangePasswordAsync(string email, string pass)
         {
-            using var db = new DatabaseContext();
-            var user = db.Users.FirstOrDefault(a => a.Email == email);
+            var user = await Database.Users.FirstOrDefaultAsync(a => a.Email == email);
 
             if (user == null)
             {
                 return false;
             }
-                
+
             var salt = HashingProcessor.CreateSalt();
             var passwordHash = HashingProcessor.GenerateHash(pass, salt);
 
             user.Password = passwordHash;
             user.Salt = salt;
-            db.SaveChanges();
+            await Database.SaveChangesAsync();
+
+            return true;
+        }
+
+        public async Task<bool> DeleteUserAsync(int userId)
+        {
+            return await DeleteUserAsync(x => x.Id == userId);
+        }
 
+        public async Task<bool> DeleteUserAsync(string email)
+        {
+            return await DeleteUserAsync(x => x.Email == email);
+        }
+
+        public async Task<bool> DeleteUserAsync(Expression<Func<UserModel, bool>> filter)
+        {
+            var temp = await Database.Users.FirstOrDefaultAsync(filter);
+            if (temp is null)
+            {
+                return false;
+            }
+            Database.Users.Remove(temp);
+            await Database.SaveChangesAsync();
             return true;
         }
```

Comment:
Where do you see this to be used?

Reply:
Remember when I was thinking where to use Events? One of the places I figured would be on user deletion, we could send out events to delete all info related to the user. Even if we don't implement that idea, this didn't really take too much time to write :D

Reply:
I understood user deletion it is usable, but I meant this exact method that takes anonymous function as parameter.

Reply:
Oh, sorry. Just figured there might be a situation when we have some specific predicates we might be removing by, so it's here just in case.